### PR TITLE
Keycloak token validation

### LINKED ISSUE #659: Proper token validation
SSO tokens should be properly validated.
Right now the token validation is based on a query to cert-giver machine.

We should instead:
 - check if certificate could be decrypted and if it is not expired
 - check if queries to user info endpoints can be made

Add also few variables with configuration of keycloak:
- realm id
- cliend id
- keycloak URL

TODO:
 - [x] token check
 - [x] keycloak configuration
 - [x] proper documentation in https://github.com/yaptide/deploy

### PR DESCRIPTION
**Pull request by p1003**

### CODE REVIEW

File: tests/test_keycloak_tokens.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import os
+from unittest.mock import patch
+
+from yaptide.routes.keycloak_routes import check_user_based_on_keycloak_token
+from werkzeug.exceptions import Forbidden, Unauthorized
+
+
+@pytest.fixture(scope='module')
+def token() -> str:
+    """Username for user with invalid password"""
```
Comment (grzanka) [inline tests/test_keycloak_tokens.py]:
```suggestion
    """Keycloak dummy token"""
```

**Review by p1003 — COMMENTED** [review]
Thx

File: tests/test_keycloak_tokens.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import jwt
+import json
```
Comment (grzanka) [inline tests/test_keycloak_tokens.py]:
```suggestion
```

File: yaptide/routes/keycloak_routes.py [hunk 3]
```diff
@@ -13,10 +15,50 @@
...
+
+        public_keys = {}
+        for jwk in jwks['keys']:
+            kid = jwk['kid']
+            public_keys[kid] = jwt.algorithms.RSAAlgorithm.from_jwk(json.dumps(jwk))
+
+        kid = jwt.get_unverified_header(token)['kid']
+        key = public_keys[kid]
+
+        _ = jwt.decode(token, key=key, audience=unverified_encoded_token["aud"], algorithms=['RS256'])
```
Comment (grzanka) [inline yaptide/routes/keycloak_routes.py]:
```suggestion
        _ = jwt.decode(token, key=key, audience=unverified_encoded_token["aud"], algorithms=['RS256'], options={"verify_signature": True})
```

Change suggested by bsobol

**Review by grzanka — APPROVED** [review]

**Pull request merged**